feature(Stock Movements to patient financia Activity)

lomamech opened the pull request:
- Addint optional parameter to displays patient stock movement

closes #4310

lomamech committed:
feature(Stock Movements to patient financia Activity)
- Addint optional parameter to displays patient stock movement

closes #4310

jniles commented on server/controllers/finance/reports/financial.patient.handlebars:
```diff
@@ -60,5 +60,32 @@
+            <tr>
+              <td>{{ add @index 1 }}</td>
+              <td><strong> {{ reference_text }} </strong></th>
+              <td> {{date date "DD/MM/YYYY HH:mm"}} </td>
```
Just FYI - if you need a timestamp, you can also use the `timestamp` filter.  Like this:
```hbs
<td> {{timestamp date }}</td>
```

jniles commented on server/controllers/medical/patients/index.js:
```diff
@@ -687,6 +688,15 @@ function getDebtorBalance(req, res, next) {
+    .catch(next);
+}
+
+function stockMovementByPatient(patientUuid) {
```
We really need more information here - it would be nice to list every item of stock that was given to the patient, not just the reference of their movement.  So, we would want:
  1. The inventory item
  2. The stock movement document reference
  2. The quantity given
  3. The type (comprimes/boite/etc)
  4. The lot number
  5. The unit price of the stock
  6. The total price of the stock.

jniles replied:
We would probably then want a total at the bottom of this with the total value of stock distributed to the patient.

jniles commented on server/controllers/finance/reports/financial.patient.js:
```diff
@@ -41,13 +43,29 @@ function build(req, res, next) {
-    .then(({ transactions, aggregates }) => {
+    .then(results => {
+      const { transactions } = results[0];
+      const { aggregates } = results[0];
```
Just FYI - you can also combine these: 
```js
const { transactions, aggregates } = results[0];
```
or 
```js
const [{transactions, aggregate}] = results;
```

This shortcut is really confusing though for people who aren't used to ECMAScript 2019.  I don't really care which way it is done, but just wanted you to be aware of it :)

jniles commented on server/controllers/finance/reports/financial.patient.js:
```diff
@@ -7,11 +8,12 @@
 
 const Patients = require('../../medical/patients');
-const Debtors = require('../../finance/debtors');
+const Debtors = require('../debtors');
```
👍

jniles commented on server/controllers/finance/reports/financial.patient.js:
```diff
@@ -41,13 +43,29 @@ function build(req, res, next) {
+        dbPromises.push(Patients.stockMovementByPatient(req.params.uuid));
+      }
+
+      return q.all(dbPromises);
```
On the server, you don't really need to use `q` anymore.  You can use `Promise.all()`.  BHIMA doesn't support any old node versions without `Promise` support.

jniles reviewed (commented):
@lomamech code looks good!

I think our friends in Vanga are going to want to see the exact items distributed to the patient, not just the reference to the stock movements.  Could you include those?  I've made comments describing the changes.

lomamech committed:
improvement(Report Patient financial Activit)
- Adding more information about item of stock givem to the patient,
- Sanitaze and improve code

lomamech committed:
Remove unused require

jniles commented on server/controllers/finance/reports/financial.patient.handlebars:
```diff
@@ -63,26 +63,59 @@
+                <tr>
+                  <td></td>
+                  <td>{{ add @index 1 }}</td>
+                  <td><strong> {{ inventory_text }} </strong></th>
```
This starts with a `<td>` and ends with a `</th>`.  Which did you want it to be?

jniles reviewed (commented):
@lomamech this looks almost perfect.  Very nicely done.

It would be nice to have a total at the bottom of all stock movements.  It would go about here:
![image](https://user-images.githubusercontent.com/896472/79743975-7c2ecc80-82fd-11ea-9457-0b3fdc18120c.png)

Once that is done, this can be merged.  It is a super helpful feature.

lomamech committed:
feature(Report Patient Standing)
- Add value of all inventory movements

jniles reviewed (approved):
Beautiful.  👍 

bors r+
